Changes to be able to build as Java 11 project

obfischer opened the pull request:
Changes to the project to be able to run it as Java 11 application.

Commit:
Changes to be able to build as Java 11 project

Changes to the project to be able to run it as Java 11 application.

StephanPirnbaum commented on pom.xml:
```diff
@@ -342,6 +369,33 @@
+                                </rules>
+                            </configuration>
+                        </execution>
+                        <execution>
```
Since no dependencies are configured here, we can drop this configuration

StephanPirnbaum replied:
execution id: enforce-banned-dependencies

obfischer replied:
Removed, commited and pushed.

DirkMahler committed:
set development version to 2.0.0-SNAPSHOT

DirkMahler commented on pom.xml:
```diff
@@ -562,10 +625,10 @@
-                            <target>1.8</target>
-                            <testSource>1.8</testSource>
-                            <testTarget>1.8</testTarget>
+                            <source>11</source>
```
This configuration already exists in lines 242ff

obfischer replied:
You are right. I will remove it. Do we still need the profile itself?

Commit:
Adjustment of the configuration of the Maven Compiler Plugin

Removal of the redundant configuration of the Maven Compiler Plugin for
the Maven profil IntelliJ.

Commit:
Execution 'enforce-banned-dependencies' removed

As we have currently no banned dependencies, the execution added for this
purpose has been removed.

Commit:
Source path setting for the JavaDoc pluin adjusted

As we use the the DeLombok plugin in all projects and from schemas
generated code very often, the JavaDoc plugin has been configured
to also process source files from the corresponding plugins for
code generation.
